sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1517,6 +1521,142 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets the application details belonging to an institution location.
+   * @param locationId location id.
+   * @param applicationId application id.
+   * @returns student application.
+   */
+  async getApplicationInfo(
+    locationId: number,
+    applicationId: number,
+  ): Promise<Application> {
+    return this.repo.findOne({
+      select: {
+        id: true,
+        data: true as unknown,
+        programYear: {
+          id: true,
+          active: true,
+          startDate: true,
+          endDate: true,
+        },
+        student: {
+          id: true,
+          birthDate: true,
+          user: {
+            id: true,
+            lastName: true,
+          },
+          sinValidation: {
+            id: true,
+            sin: true,
+          },
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+          },
+        },
+      },
+      relations: {
+        programYear: true,
+        location: true,
+        student: {
+          user: true,
+          sinValidation: true,
+        },
+        currentAssessment: {
+          offering: true,
+        },
+      },
+      where: {
+        id: applicationId,
+        location: {
+          id: locationId,
+        },
+        applicationStatus: Not(ApplicationStatus.Overwritten),
+      },
+    });
+  }
+
+  /**
+   * Set of validations for existing application and newly
```

Comment:
Just a thought. As this class(ApplicationService) is having more than 1500 lines of code already, should we consider creating a service like `ApplicationOfferingService` and move this method there?

Reply:
I also had similar thought when I was doing this, but this validation is not specific to `Application Offering change request`, so I thought its better to do it in `application.service`